MapleServer2/Types/Player.cs

```diff
@@ -118,14 +118,29 @@ public class Player
         private Task HpRegenThread;
         private Task SpRegenThread;
         private Task StaRegenThread;
-
+        private Task OnlineDurationThread;
+        private TimeInfo Timestamps;
         public Dictionary<int, PlayerStat> GatheringCount = new Dictionary<int, PlayerStat>();
 
+        class TimeInfo
+        {
+            public long charCreation = -1;
+            public int onlineDuration = 0;
+            public long lastOnline = -1;
```

Comment:
You don't need these defaults here if you're already setting them in the constructor parameters